### DISCUSSION TITLE
is volumeMode: Block supported?

### DISCUSSION THREAD

**Discussion by clrfuerst** [backube/volsync#495] [Q&A] [answered]
This is a great utility, but trying to sync an RPD PVC (volumeMode: Block) from one cluster to another and the Destination Cluster container won't start because the volumeMode is Block.

container has this as an event.

Unable to attach or mount volumes: unmounted volumes=[data], unattached volumes=[data keys kube-api-access-hxmll]: volume data has volumeMode Block, but is specified in volumeMounts.

Is this something supported and I have configurations wrong?  Thanks

**Comment by JohnStrunk** [comment, marked as answer]
VolSync doesn't currently support `volumeMode: block` volumes. The main reason that we haven't added support is that there's no efficient way to calculate a diff so that we can send just the changed blocks during each sync iteration.
There is upstream Kubernetes work that is hoping to provide a standard way of exposing changed blocks. Once that happens, we plan to add block volume support to VolSync.
See: kubernetes/enhancements#3367
